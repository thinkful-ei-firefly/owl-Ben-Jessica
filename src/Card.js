import React from 'react';
import './card.css';

function Card(props) {
    return (
      
        <div id={props.id} className='card'>
            <p className='stageName'>{props.name}</p>
            <img src={props.avatar} className='stageIcon' alt='stage icon' />
          
        </div>
     
    );
  }
  
  export default Card;