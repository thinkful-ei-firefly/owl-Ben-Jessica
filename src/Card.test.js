import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card';

describe('Card component', () => {
  const card = '<Card id=1 name="Test name" avatar="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1" />';
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Card />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(card)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders the UI as expected with no unreads', () => {
    const tree = renderer
      .create(card)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});