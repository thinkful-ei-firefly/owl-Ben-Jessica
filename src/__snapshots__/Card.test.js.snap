// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the UI as expected 1`] = `"<Card id=1 name=\\"Test name\\" avatar=\\"https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1\\" />"`;

exports[`Card component renders the UI as expected with no unreads 1`] = `"<Card id=1 name=\\"Test name\\" avatar=\\"https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1\\" />"`;
